### PR TITLE
Add Solid Torrents engine

### PR DESCRIPTION
**Pull request by mikeri**
## What does this PR do?
Add Solid Torrents engine

## Why is this change important?
It's not

## How to test this PR locally?
Comment out "disabled : True" from the solidtorrents config section

### CODE REVIEW

**Comment by return42** [comment]
@maiki thanks for contributing .. do you now if there exists a documentation of the API (`https://www.solidtorrents.net/api/v1`)?

**Review by return42 — CHANGES_REQUESTED** [review]
LGTM .. I have only some minor nits .. you can fetch from my branch: https://github.com/return42/searx/commits/solidtorrents

**Comment by kvch** [comment]
After the comments above are addressed, I will approve the PR.

**Comment by mikeri** [comment]
@return42, I don't know about any documentation. Thanks for the fixes!

**Comment by return42** [comment]
@mikeri since last patches on branch master there is a conflict in the `./Makefile` .. can you solve this conflict / or fetch again from my branch where I solved the conflict already. / thanks!

**Comment by mikeri** [comment]
@return42 done.

**Comment by unixfox** [comment]
> @mikeri since last patches on branch master there is a conflict in the `./Makefile` .. can you solve this conflict / or fetch again from my branch where I solved the conflict already. / thanks!

mmh why is it not a separate PR for that instead? it's unrelated to this new engine.

**Comment by return42** [comment]
> mmh why is it not a separate PR for that instead? it's unrelated to this new engine.

sorry for being misleading / I was asking for a rebase on branch master .. nothing special.

**Comment by mikeri** [comment]
Sorry for not being very experienced with git (I mostly use it for single user use), what exactly do I need to do?

**Comment by dalf** [comment]
Can you squash your commit ?

```
git rebase -i HEAD~2
```

Then on the second commit (f484a7cc55f4791523ea4b7e6c4eb956854934a0) , replace `pick` by `fixup` 

Save, check the two commits are merged : `git log`.

Then push force on your branch:

```
git push --force mikeri solidtorrents
```

(may be it is not `mikeri`, get the right value using `git remote -v`)

----

Instead of `git merge origin master` you can `git pull --rebase origin master` : it will rewrite your git history in your branch, but you won't have a merge commit ( f484a7cc55f4791523ea4b7e6c4eb956854934a0 ).

**Comment by return42** [comment]
@dalf just a suggestion: By github defaults, maintainers  have write access to the branch of a PR. More and more we have discussions about git force and squash .. most often it consumes a lot of time to teach the sender of a PR how to process such tasks. If you think it takes to much time in one ore the other case, feel free to make use of the write access.

**Comment by mikeri** [comment]
I tried @dalf 's advice (`git pull --rebase origin master`) and pushing, but that polluted the PR with all the commits. I tried undoing it and then force pushing with the Makefile conflict fixed, but the PR still seems polluted with my mistake. Should I just make a new PR?

**Comment by return42** [comment]
> Should I just make a new PR?

No need for .. lets wait .. we will find a solution for ... :)

**Comment by return42** [comment]
> I tried @dalf 's advice (git pull --rebase origin master) and pushing, but that polluted the PR with all the commits.

I guess that did not worked, since your "origin" is your fork and not the searx repository.  You need to add searx reposetory as a another remote (name it  "upstream") and then you can do a `git pull --rebase upstream master`

For convenience  I did it for you, see my branch: https://github.com/return42/searx/commits/solidtorrents

You or even @dalf can fetch from and push on the solidtorrents branch of your searx fork.  /  I'm not a maintainer, so I do not have write access to your branch :)

**Comment by mikeri** [comment]
I recreated the branch, it should be good to go now?

**Comment by return42** [comment]
LGTM / thanks.

TLDR; 

> I recreated the branch, 

I recommend to teach yourself git workflows .. it's worth .. git is the SCM you will work with in the next 20 years ;-)

**Review by return42 — APPROVED** [review]

**Review by kvch — APPROVED** [review]

**Review by unixfox — APPROVED** [review]
